MMC boot partition

Allow our kernel to configure without exposing the MMC boot
partitions.
Configures added:
MMC_BLOCK_MMC_PART
MMC_BLOCK_MMC_PART_BOOT (wrapper)

This patch is based on Andrei Warkentin's patch "[RFC] MMC: MMC boot partitions support."

Signed-off-by: Theodor Gherzan <theodor@resin.io>
---
drivers/mmc/card/Kconfig   |   18 +++++
drivers/mmc/card/block.c   |    2 ++
2 files changed, 20 insertions(+), 0 deletions(-)
diff --git a/drivers/mmc/card/Kconfig b/drivers/mmc/card/Kconfig
index 5562308..6c21809 100644
--- a/drivers/mmc/card/Kconfig
+++ b/drivers/mmc/card/Kconfig
@@ -14,6 +14,24 @@ config MMC_BLOCK
 	  mount the filesystem. Almost everyone wishing MMC support
 	  should say Y or M here.

+config MMC_BLOCK_MMC_PART
+   tristate "MMC block MMC device partitioning support"
+   depends on BLOCK
+   default n
+   help
+     Say Y here to enable access to MMC partitions other than
+     the user area, such as the boot partitions. The other partitions
+     will appear as separate mmcblk devices.
+
+config MMC_BLOCK_MMC_PART_BOOT
+   tristate "MMC block MMC boot partition support"
+   depends on MMC_BLOCK_MMC_PART
+   default n
+   help
+     Say Y here to enable access to the MMC boot partitions. Boot
+     partitions are used by some devices to store bootloader information.
+     The boot partitions will appear as separate mmcblk devices.
+
 config MMC_BLOCK_MINORS
 	int "Number of minors per block device"
 	depends on MMC_BLOCK
diff --git a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
index 1137130..84eaeb0 100644
--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -2296,6 +2296,7 @@ static int mmc_blk_alloc_parts(struct mmc_card *card, struct mmc_blk_data *md)
 	if (!mmc_card_mmc(card))
 		return 0;

+#ifdef CONFIG_MMC_BLOCK_MMC_PART_BOOT
 	for (idx = 0; idx < card->nr_parts; idx++) {
 		if (card->part[idx].size) {
 			ret = mmc_blk_alloc_part(card, md,
@@ -2308,6 +2309,7 @@ static int mmc_blk_alloc_parts(struct mmc_card *card, struct mmc_blk_data *md)
 				return ret;
 		}
 	}
+#endif

 	return ret;
 }
